lab-caleb/lib/index.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+module.exports = exports = {};
+
+exports.products = {};
+exports.products.gfxcard = [];
+exports.products.keyboard = [];
+
+
+exports.Product = function(name, price, sku){
+  this.addToProducts = function(){
+    exports.products[this.category].push(this);
+  };
+  this.name = name;
+  this.price = price;
+  this.sku = sku;
+};
+
+exports.Card = function(name, price, sku, architecture, category){
+  exports.Product.call(this, name, price, sku);
+  this.architecture = architecture;
+  this.category = category;
+};
+
+exports.Keyboard = function(name, price, sku, backlighting, category){
+  exports.Product.call(this, name, price, sku);
+  this.backlighting = backlighting;
+  this.category = category;
+};
+
```

Comment:
**TODO FOR LOST POINTS**
This all looks great, but any instantiation and function invocation should happen in an ``index.js`` file that sits in the root of your folder, i.e., it shares the same level as your ``.gitignore``. Any modules sitting in the ``lib`` folder are only where you define and declare your functions.